### PR TITLE
fix(input): input with file type

### LINKED ISSUE #2311: [BUG] - Bug when input type is file.
### NextUI Version

~~2.2.9~~ 2.4.1

### Describe the bug

When using the input component with file type, I observed that on the first click to choose a file, the selected file is logged as "undefined" in the console. However, on the second attempt, the correct file information is logged

### Your Example Website or App

_No response_

### Steps to Reproduce the Bug or Issue

Click the "Choose File" button to open the file dialog.
On the first click, notice that the selected file is logged as "undefined" in the console.
On the second click, the correct file information is logged.

### Expected behavior

The file input value should be cleared consistently on each click to ensure that the correct file information is logged on the first attempt.

### Screenshots or Videos

https://github.com/nextui-org/nextui/assets/95061742/e210cb3e-c389-49ec-99b6-469d68d7a86b

### Operating System Version

windows 11

### Browser

Edge

### LINKED ISSUE #2397: [BUG] - Input with type file not working on first attempt
### NextUI Version

~~2.2.9~~ 2.4.1

### Describe the bug

Adding an Input component with type file requires you to select the file twice before it sets it as value.

### Your Example Website or App

_No response_

### Steps to Reproduce the Bug or Issue

1. Latest NextJS with App Router (fresh install)
2. Add a <Input type="file" name="file" /> to any page
3. Select a file, you will see the value does not update and it sill says No file chosen
4. Select a file again, now it updates and sets the file

### Expected behavior

Should set the file on first attempt

### Screenshots or Videos

Screen Recording 2024-02-21 at 23 46 27

### Operating System Version

MacOS

### Browser

Chrome

### LINKED ISSUE #2965: Input for file type to upload images single or multiple
### Is your feature request related to a problem? Please describe.

I need to have a file Input type for uploading images

### Describe the solution you'd like

As a user, I want to click on an input element and select one or multiple images for uploading.

### Describe alternatives you've considered

I am using the Input element with type as a file.

```
<Input type="file" />
```

### Screenshots or Videos

Screenshot (220)

### PR DESCRIPTION
**Pull request by wingkwong**
- Closes #2397
- Closes #2311
- Closes #2965

## 📝 Description

- Fixed "uploading file only works on the second attempt" issue
- Revised styles for input[type="file"]
- Revised labelPlacement logic (only support `outside` / `outside-left`) for input[type="file"]

## ⛳️ Current behavior (updates)

image

## 🚀 New behavior

image

image

image

## 💣 Is this a breaking change (Yes/No):

No

## Summary by CodeRabbit

- **New Features**
  - Introduced support for file type inputs in the input component, enabling seamless file uploads.

- **Style**
  - Updated styles to ensure file input elements have a pointer cursor and transparent background.

- **Bug Fixes**
  - Adjusted logic to prevent blocking of file uploads by removing `value` and `onChange` attributes for file inputs. 

- **Documentation**
  - Updated input storybook to include examples of file input types.

### CODE REVIEW

File: packages/components/input/src/use-input.ts [hunk 1]
```diff
@@ -212,6 +221,22 @@ export function useInput<T extends HTMLInputElement | HTMLTextAreaElement = HTML
...
+    if (isFileTypeInput) {
+      // if `labelPlacement` is not defined, choose `outside` instead
+      // since the default value `inside` is not supported in file input
+      if (!originalProps.labelPlacement) return "outside";
+
+      // throw a warning if `labelPlacement` is `inside`
+      // and change it to `outside`
+      if (originalProps.labelPlacement === "inside") {
+        /* eslint-disable no-console */
+        console.warn(
```
Comment (jrgarciadev) [inline packages/components/input/src/use-input.ts:233]: @wingkwong  use `warn` function from `"@nextui-org/shared-utils";`

Comment (wingkwong) [reply packages/components/input/src/use-input.ts]: Good to know!

**Comment by jrgarciadev** [comment]
@wingkwong add the `file` type to the `types` storybook https://nextui-storybook-v2-git-fix-input-file-type-nextui-org.vercel.app/?path=%2Fstory%2Fcomponents-input--input-types

**Comment by wingkwong** [comment]
@jrgarciadev added to Input Types.

image

**Review by jrgarciadev — APPROVED** [review]

**Pull request merged**